Updates tests to work on v23.0

zzantares opened the pull request:
When running `cabal test all` I was observing a few tests failing, this is mostly due to changes in default behavior on v23.0; this updates the test and they now work fine.

zzantares committed:
Fixes wallet tests wrt dumpwallet RPC command

The tests assumed that a legacy wallet was created with the `createwallet` RPC
command, but since v23.0 the command creates native descriptor wallet which is
not compatible with the `dumpwallet` command. Since the `dumpwallet` command
only works for legacy wallets it makes sense to always create a legacy wallet
for in the tests for it.

zzantares committed:
Fixes tests wrt listdescriptors RPC command

For bitcoind v23.0 and up there are a couple more descriptors automatically
added for new wallets, and that was causing the tests to fail on the new version.

GambolingPangolin commented on bitcoind-regtest/test/Bitcoin/Core/Test/Wallet.hs:
```diff
@@ -251,7 +251,9 @@ testDescriptorCommands v = do
-        when (v > v21_1) $ RPC.listDescriptors >>= shouldMatch 6 . length
+        when (v > v21_1 && v < v23_0) $ RPC.listDescriptors >>= shouldMatch 6 . length
+        -- Newly created descriptor wallets will contain an automatically generated tr() descriptor
+        when (v > v23_0) $ RPC.listDescriptors >>= shouldMatch 8 . length
```
Should this be `>=`?  or perhaps the above should be `<=`?

zzantares replied:
Yes good catch! I've updated it 👍🏼.

Fixed in 16bc34f291b7

zzantares committed:
Cover case when testing listing descriptors against v23_0

When using v23.0 exactly the list descriptors test wasn't run.

zzantares committed:
Pin and format using the default fourmolu configuration

This adds a `fourmolu.yaml` and formats the code accordingly, it was generated using:

``` console
$ fourmolu --print-defaults > fourmolu.yaml
```

The motivation being that we have this in the `README.md`:

> Please use `ormolu` or `fourmolu` (with the default configuration) for code formatting.

However, for users that use a different configuration at the user level
configuration this is inconvenient since there's not an easy way of using the
default configuration.

wraithm committed:
Merge pull request #19 from bitnomial/pin-fourmolu-config

Pin and format using the default fourmolu configuration
